src/legacy/core_plugins/management/public/np_ready/services/index_pattern_management/list/config.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { StaticIndexPattern, Field } from '../../../../../../data/public';
```

Comment:
We have [deprecated `StaticIndexPattern` and replaced with `IIndexPattern`](https://github.com/elastic/kibana/blob/3a915b22a04dc85e257a1cc8802178dc541d83ca/src/legacy/core_plugins/data/public/index_patterns/index_patterns/index_pattern.ts#L49-L53) -- can we switch these over to that interface instead?

https://github.com/elastic/kibana/blob/3a915b22a04dc85e257a1cc8802178dc541d83ca/src/legacy/core_plugins/data/public/index_patterns/index_patterns/index_pattern.ts#L49-L53

Reply:
I did a little investigation and found out that `IIndexPattern` isn't exposed in `legacy/core_plugins/data`, but in `src/plugins/data/public`. So, I used it. 

Also, I changed `Field` to `IFieldType` there. Because plugins like `rollup` can extend the type.